### PR TITLE
feat/init frontend

### PR DESCRIPTION
**Pull request by nguyenvukhang**
typeorm has got to speed up their documentation
next.js api doesn't play well with typeorm's fancy decorators
always try to normalize things before connecting it with something else
big thanks to @weiseng18 for inspiration that led to this particular implementation

### CODE REVIEW

**Comment by nguyenvukhang** [comment]
to test, create an `.env.local` at `apps/web`, with `NEXT_PUBLIC_BACKEND=http://localhost:8080`

run `yarn dev` from root and go to `localhost:3000/search`

File: apps/web/pages/search.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { ReactNode, useEffect, useState } from 'react'
+import Search from '../components/Search'
+import { H1 } from '../components/Html'
+
+export default function SearchPage() {
+  const queryState = useState('')
+  const [query, setQuery] = queryState
+  const [results, setResults] = useState<string[]>([])
+  const [ResultsDisplay, setResultsDisplay] = useState<ReactNode[]>([])
```
Comment (weiseng18) [inline apps/web/pages/search.tsx:9]: do you want to lowercase this?

File: packages/database/src/controller/ModuleCondensedController.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { AppDataSource } from '../data-source'
+import { Request, Response } from 'express'
+import { ModuleCondensed } from '../entity-repo/ModuleCondensed'
+import { Like } from 'typeorm'
+
+/** ModuleCondensed api controller */
+export class ModuleCondesnedController {
```
Comment (weiseng18) [inline packages/database/src/controller/ModuleCondensedController.ts:7]: typo in class name

File: packages/database/src/routes.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { ModuleCondesnedController } from './controller/ModuleCondensedController'
```
Comment (weiseng18) [inline packages/database/src/routes.ts:1]: same in this file

File: apps/web/pages/search.tsx (hunk 1 above)
Comment (nguyenvukhang) [reply apps/web/pages/search.tsx:9]: It's used as a JSX component, so no

**Review by weiseng18 — APPROVED** [review]
nice

**Pull request merged**